src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1275,6 +1275,15 @@ void Lowering::LowerHWIntrinsicCmpOp(GenTreeHWIntrinsic* node, genTreeOps cmpOp)
         node->gtOp1 = op1;
         BlockRange().Remove(op2);
 
+        op2 = op2->AsOp()->gtGetOp1();
+
+        if (op2 != nullptr)
+        {
+            // Some zero vectors are Create/Initialization nodes with a constant zero operand
+            // We should also remove this to avoid unused value assertions
+            BlockRange().Remove(op2);
```

Comment:
Assert `op2->IsIntegralConst(0)`?

Edit: also nit: I don't think that avoiding assertions is the goal here. The goal is to just remove dead code.

Reply:
There isn't a need to assert IMO, this is under an `if (op2->IsIntegralConstVector(0))` check